napari/components/_tests/test_layer_slicer.py

```diff
@@ -244,3 +245,37 @@ def test_slice_layers_async_task_to_layers_lock(layer_slicer):
 
     assert task.result()[layer].id == 1
     assert task not in layer_slicer._layers_to_task
+
+
+def test_slice_layers_exception_main_thread(layer_slicer):
+    """Exception is raised on the main thread from an error on the main
+    thread immediately when the task is created."""
+
+    class FakeAsyncLayerError(FakeAsyncLayer):
+        def _make_slice_request(self, dims) -> FakeSliceRequest:
+            assert current_thread() == main_thread()
```

Comment:
Not sure the detail of which thread this method runs on is too important here and in the other test, so I would consider removing it.

Reply:
you're right, I'll clean that up.